### PR TITLE
CMR-9085: Add support for umm-T to CMR graphql

### PR DESCRIPTION
**Pull request by dmistry1**
# Overview
Add support for umm-T to CMR graphql
### What is the feature?

Adding support for UMM-T draft
Updated mmtQuery.js to use the MMT new API
Update collection draft retrieval to use new MMT API and updated test cases.
Added test cases for 
- cmr/concept/toolDraft.js
- datasource/toolDraft.js
- resolver/toolDraft.js

### CODE REVIEW

File: src/datasources/__tests__/collectionDraft.test.js [hunk 1]
```diff
@@ -61,7 +63,7 @@ describe('collectionDraft', () => {
       }
     }, {
       headers: {
-        'Client-Id': 'eed-test-graphql',
+        'Client-Id': 'mmt-react-ui',
```
Comment (cgokey) [inline src/datasources/__tests__/collectionDraft.test.js:66]: I think the client-id shouldn't change.   The client isn't a react ui, it is  a eed test graph client.

File: src/resolvers/__tests__/collectionDraft.test.js [hunk 2]
```diff
@@ -28,9 +28,11 @@ describe('Collection', () => {
             .defaultReplyHeaders({
               'X-Request-Id': 'abcd-1234-efgh-5678'
             })
-            .get(/collection_drafts/)
+            .get(/api/)
```
Comment (cgokey) [inline src/resolvers/__tests__/collectionDraft.test.js:31]: shouldn't the endpoint be /api/drafts now?

File: src/cmr/concepts/collectionDraft.js [hunk 3]
```diff
@@ -84,7 +85,7 @@ export default class CollectionDraft extends Concept {
 
     // Construct the promise that will request data from the umm endpoint
     return mmtQuery({
-      conceptType: this.getConceptType(),
+      draftType: 'CollectionDraft',
```
Comment (cgokey) [inline src/cmr/concepts/collectionDraft.js:88]: Shouldn't this say this.getConceptType() still ?

Comment (dmistry1) [reply src/cmr/concepts/collectionDraft.js]: Agree, I changed it back to this.getConceptType(). Originally hard coded them for testing purpose, forgot to change it back.

File: src/resolvers/__tests__/collectionDraft.test.js (hunk 2 above)
Comment (dmistry1) [reply src/resolvers/__tests__/collectionDraft.test.js]: Updated to `/api\/drafts/`.

**Review by tle1989 — APPROVED** [review]

**Review by sxu123 — APPROVED** [review]

File: src/cmr/concepts/toolDraft.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  pick
+} from 'lodash'
+import snakecaseKeys from 'snakecase-keys'
+
+import { mergeParams } from '../../utils/mergeParams'
+import { mmtQuery } from '../../utils/mmtQuery'
+import { parseError } from '../../utils/parseError'
+import Concept from './concept'
+
+export default class ToolDraft extends Concept {
```
[the remark this answers is not in the document]
Comment (macrouch) [reply src/cmr/concepts/toolDraft.js:13]: Actually, maybe we don't need `ToolDraft` and `CollectionDraft` if the only difference between them is that hardcoded type in the constructor. Maybe we just use `DraftConcept` and add the `conceptType` as a parameter that can be passed in from the initialization in the datasource?

File: src/utils/umm/toolDraftKeyMap.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "associationDetails": "meta.association-details",
+    "conceptId": "meta.concept-id",
+    "contactGroups": "ContactGroups",
+    "contactPersons": "ContactPersons",
+    "description": "Description",
+    "doi": "DOI",
+    "longName": "LongName",
+    "metadataSpecification": "MetadataSpecification",
+    "name": "Name",
+    "nativeId": "meta.native-id",
```
[the remark this answers is not in the document]
Comment (dmistry1) [reply src/utils/umm/toolDraftKeyMap.json]: Correct, meta does not come back in the MMT response. I removed those fields

File: src/types/toolDraft.graphql [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  "This is the contact persons of the downloadable tool or web user interface."
+  contactPersons: JSON
+  "A brief description of the web user interface or downloadable tool. Note: This field allows lightweight markup language with plain text formatting syntax. Line breaks within the text are preserved."
+  description: String
+  "The Digital Object Identifier (DOI) of the web user interface or downloadable tool."
+  doi: String
+  "The native id of a tool."
+  nativeId: String
+  "This element describes the latest date when the tool was most recently pushed to production for support and maintenance. "
+  lastUpdatedDate: String
```
[the remark this answers is not in the document]
Comment (dmistry1) [reply src/types/toolDraft.graphql:21]: Added LastUpdatedDate field into toolDraft and tool keymaps.json

File: src/cmr/concepts/toolDraft.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (dmistry1) [reply src/cmr/concepts/toolDraft.js]: Okay so I created a DraftConcept.js which now take in an additional parameter called conceptType. That param in being passed into from datasource. With this implementation we can use DraftConcept.js for all draftTypes by passing in the conceptType in the datasource. Therefore removed src/cmr/concepts/CollectionDraft.js and src/cmr/concepts/ToolDraft.js as there are not needed. 🙂

File: src/types/tool.graphql [hunk 7]
```diff
@@ -17,6 +17,8 @@ type Tool {
   doi: String
   "The native id of a tool."
   nativeId: String
+  "This element describes the latest date when the tool was most recently pushed to production for support and maintenance. "
```
[the remark this answers is not in the document]
Comment (dmistry1) [reply src/types/tool.graphql:20]: ✅

File: src/cmr/concepts/draftConcept.js [hunk 8]
```diff
@@ -11,15 +11,17 @@ import { parseError } from '../../utils/parseError'
 
 import Concept from './concept'
 
-export default class CollectionDraft extends Concept {
+export default class DraftConecpt extends Concept {
```
Comment (macrouch) [inline src/cmr/concepts/draftConcept.js:14]: misspelled class name

File: src/cmr/concepts/draftConcept.js [hunk 9]
```diff
@@ -11,15 +11,17 @@ import { parseError } from '../../utils/parseError'
...
   /**
    * Instantiates a Collection object
    * @param {Object} headers HTTP headers provided by the query
    * @param {Object} requestInfo Parsed data pertaining to the Graph query
    * @param {Object} params GraphQL query parameters
+   * @param {Object} conceptType Passed in conceptType from DataSoruce
```
Comment (macrouch) [inline src/cmr/concepts/draftConcept.js:20]: misspelled DataSource

**Review by macrouch — APPROVED** [review]

**Pull request merged**